x/evm/keeper/statedb.go

```diff
@@ -1,265 +1,589 @@
 package keeper
 
 import (
+	"bytes"
+	"fmt"
 	"math/big"
 
-	"github.com/cosmos/ethermint/x/evm/types"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/crypto"
+	tmtypes "github.com/tendermint/tendermint/types"
 
+	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-
-	ethcmn "github.com/ethereum/go-ethereum/common"
-	ethstate "github.com/ethereum/go-ethereum/core/state"
-	ethtypes "github.com/ethereum/go-ethereum/core/types"
-	ethvm "github.com/ethereum/go-ethereum/core/vm"
+	ethermint "github.com/cosmos/ethermint/types"
+	"github.com/cosmos/ethermint/x/evm/types"
 )
 
+var _ vm.StateDB = &Keeper{}
+
+// csdb defines the internal field values used on the StateDB operations
+type csdb struct {
+	// The refund counter, also used by state transitioning.
+	refund uint64
+
+	// Per-transaction access list
+	accessList *types.AccessListMappings
+
+	// Transaction counter in a block. Used on StateSB's Prepare function.
```

Comment:
Comment should say something on the lines of how this is an index for the transaction.

We keep track of this to see what we've processed in terms of state for the block (i.e. populating fields from the log)